mula/scheduler/connectors/services/katalogus.py

```diff
@@ -43,91 +43,97 @@ def __init__(self, host: str, source: str, timeout: int, pool_connections: int,
         self.flush_caches()
 
     def flush_caches(self) -> None:
-        self.flush_plugin_cache()
-        self.flush_normalizer_cache()
-        self.flush_boefje_cache()
-
-    def flush_plugin_cache(self) -> None:
+        plugins = self.flush_plugin_cache()
+        self.flush_boefje_cache(plugins)
+        self.flush_normalizer_cache(plugins)
+    
+    def flush_plugin_cache(self) -> dict[str, dict]:
         self.logger.debug("Flushing the katalogus plugin cache for organisations")
 
+        orgs = self.get_organisations()
+        plugin_cache = {}
+        boefjes_cache = {}
+        for org in orgs:
+            plugin_cache.setdefault(org.id, {})
+            plugins = self.get_plugins_by_organisation(org.id)
+            plugin_cache[org.id] = {plugin.id: plugin for plugin in plugins if plugin.enabled}
+
         with self.plugin_cache_lock:
             # First, we reset the cache, to make sure we won't get any ExpiredError
             self.plugin_cache.expiration_enabled = False
             self.plugin_cache.reset()
-
-            orgs = self.get_organisations()
-            for org in orgs:
-                self.plugin_cache.setdefault(org.id, {})
-
-                plugins = self.get_plugins_by_organisation(org.id)
-                self.plugin_cache[org.id] = {plugin.id: plugin for plugin in plugins if plugin.enabled}
-
+            self.plugin_cache = plugin_cache
             self.plugin_cache.expiration_enabled = True
-
+        
         self.logger.debug("Flushed the katalogus plugin cache for organisations")
+        return plugins
```

Comment:
If I'm not mistaken the plugins from only the last organisation might only get returned

Reply:
yes, good catch, should have been the temp plugin_cache dict, not the innner plugins dict.